make it extensible and add support for C/C++

tizee opened the pull request:
Thanks for your work.

Here is what includes in this PR.

1.  Make it more extensible by reusing the shared logic for swapping nodes.

We can add more processors for languages that support the ternary expression. 

2. Add support for C and C++

I've also added tests by reusing some test cases from `js`. BTW, This can be further optimized since all languages with ternary expression look similar.

3. Remove `javascriptreact` in `tests/init.lua`.

I believe it has been included in the `tree-sitter-javascript` parser.

4. Add a VimScript function `swap_ternary#swap()`.

I think it's more easier to type than `require('swap-ternary').start()`. The `swap_ternary#swap()` is merely a wrapper of `require('swap-ternary').start()`.

5. Polish up README

Add instructions on how to setup a keybinding for this plugin.

6. Add `plugin/swap_ternary.lua`

The `plugin/swap_ternary.lua` is an indicator for me to test whether this plugin has been loaded. It's up to you to decide whether to add this file.

tizee committed:
refactor: make it extensible

- support for C and C++

tizee committed:
update readme

xlboy commented:
I am very sorry that my English is not very good. I translated these through the translation software. Please forgive me if I have caused you a bad understanding

---

Thank you very much for your PR! Actually, adding other languages is also in my plan.

1. OK. However, there may be some additional special cases, such as the node structure of `python` being different from `c, java, js`, but it does not affect this PR :)

2. OK.

    It also needs to take care of the special cases mentioned above (`python`).

    Regarding "further optimization," I feel that it is not necessary because the relevant test cases are generated by AI. If we further abstract and optimize, it may lose its simple and clear meaning.

3. OK.

4. OK, let's just replace the existing `start` with `swap`! (In the early stages, only my friends and I were using it, so it could be adjusted quickly)

5. OK.

6. For the sake of simplicity, I feel it is not necessary :)

tizee committed:
revert plugin since it's not needed

tizee committed:
rename start to swap for invoking

tizee commented:
No worries, I can get your idea. 
I've removed the unnecessary `plugin/swap-terneray.lua` and renamed `start` to `swap` for invoking with `:call require('swap-internary').swap()`. Any more suggestions?

xlboy commented:
Thank you for your understanding.

I have no other suggestions, and I appreciate your contribution! Wish you a happy life~
